data/src/main/java/ca/bc/gov/data/datasource/remote/OrganDonorRemoteDataSource.kt

```diff
@@ -0,0 +1,46 @@
+package ca.bc.gov.data.datasource.remote
+
+import android.util.Base64
+import ca.bc.gov.common.const.MESSAGE_INVALID_RESPONSE
+import ca.bc.gov.common.const.SERVER_ERROR
+import ca.bc.gov.common.exceptions.MyHealthException
+import ca.bc.gov.common.model.services.OrganDonationDto
+import ca.bc.gov.data.datasource.remote.api.HealthGatewayPrivateApi
+import ca.bc.gov.data.datasource.remote.model.base.patientdata.PatientDataType
+import ca.bc.gov.data.model.mapper.toDto
+import ca.bc.gov.data.utils.safeCall
+import javax.inject.Inject
+
+/**
+ * @author Pinakin Kansara
+ */
+class OrganDonorRemoteDataSource @Inject constructor(
+    private val healthGatewayPrivateApi: HealthGatewayPrivateApi
+) {
+
+    /**
+     * This API is used to get organ donor data based on type
+     * @see [PatientDataType]
+     * uses api = v2
+     */
+    suspend fun getOrganDonorData(hdid: String, token: String): OrganDonationDto {
+        val response = safeCall {
+            healthGatewayPrivateApi.getPatientData(
+                hdid,
+                token,
+                listOf(PatientDataType.ORGAN_DONOR.value)
+            )
+        }
```

Comment:
Instead of items[0], could you use firstOrNull in the safeCall, so if the list is empty it'll also throw a MyHealthException (instead of array out of bounds exception)

Reply:
updated